spec/inputs/kinesis_spec.rb

```diff
@@ -38,6 +38,50 @@
     "initial_position_in_stream" => "LATEST"
   }}
 
+  # Config hash to test valid additional_kcl_options
+  let(:config_with_valid_additional_kcl_options) {{
+    "application_name" => "my-processor",
+    "kinesis_stream_name" => "run-specs",
+    "codec" => codec,
+    "metrics" => metrics,
+    "checkpoint_interval_seconds" => 120,
+    "region" => "ap-southeast-1",
+    "profile" => nil,
+    "additional_kcl_options" => {
+        "initial_lease_table_read_capacity" => 25,
+        "initial_lease_table_write_capacity" => 100,
+        "kinesis_endpoint" => "http://www.localhost"
+    }
+  }}
+
+  # Config hash to test invalid additional_kcl_options where the name is not found
+  let(:config_with_invalid_additional_kcl_options_name_not_found) {{
+    "application_name" => "my-processor",
+    "kinesis_stream_name" => "run-specs",
+    "codec" => codec,
+    "metrics" => metrics,
+    "checkpoint_interval_seconds" => 120,
+    "region" => "ap-southeast-1",
+    "profile" => nil,
+    "additional_kcl_options" => {
+        "foo" => "bar"
+    }
+  }}
+
+  # Config hash to test invalid additional_kcl_options where the type is complex or wrong
+  let(:config_with_invalid_additional_kcl_options_wrong_type) {{
```

Comment:
This config appears to be unused

Reply:
Fixed by actually using it in a new test :)